### ISSUE TITLE
Extremely poor range using rtl8723be

### ISSUE THREAD

**Issue by dean-brown** [lwfinger/rtlwifi_new#101]
I have a hp-pavillion laptop. 
Windows drivers work fine with the rtl8723be card. 

I have tried both the master branch and the test branch and neither have resolved my issue. 
I have "options rtl8723be fwlps=0 swlps=0 ant_sel=2" in my  "/etc/modprobe.d/50-rtl8723be.conf"

Steps followed:
make 
sudo make install
sudo modprobe -rv rtl8723be
sudo modprobe -v rtl8723be ant_sel=2
sudo ip link set wlp2s0 up
sudo iw dev wlp2s0 scan    # test the device 

sudo echo "options rtl8723be fwlps=0 swlps=0 ant_sel=2" > /etc/modprobe.d/50-rtl8723be.conf

Reboot (just to make sure!)

The connection is reliable. Which is a vast improvement to how it was before installation of the new driver. But if my router is more than 2 - 3 meters away it can't be found. 

I understand that some people have had success with this but I have followed instructions to the letter and still have issues with range.

**Comment by lwfinger** [comment]
You have been trying the wrong branch. You need rock.new_btcoex, otherwise the ant_sel module parameter does not work.

**Comment by dean-brown** [comment]
I actually love you! 
A week i've been at this with no success. I downloaded the zip to be sure I was using the correct branch.

Works flawlessly. You also thought me something about git. Apparently the command i have been using to clone branches just clones the master branch >< 

Thanks :+1:

**Comment by lwfinger** [comment]
You should investigate the "git checkout" command.

**Comment by krlk89** [comment]
@lwfinger Will the driver in the rock.new_btcoex branched be added to the Linux kernel in the (near) future?

**Comment by lwfinger** [comment]
I have answered this question several times already, but the short answer is that it is not easy to port that fix. Realtek completely rewrote the btcoex routines to use interprocess communication methods that will never be accepted into the mainline kernel.

**Comment by chembai** [comment]
i cannot run the make command in my kali linux (hp -pavillion laptop )
it gives error:- 
make -C /lib/modules/4.3.0-kali1-amd64/build M=/root/rtlwifi_new modules
make[1]: *** /lib/modules/4.3.0-kali1-amd64/build: No such file or directory.  Stop.
Makefile:57: recipe for target 'all' failed
make: *** [all] Error 2

btw sorry im new to linux environment.

**Comment by lwfinger** [comment]
You have not installed the kernel modules. You will need to find out how to do that from the Kali forum. The method is different for each distro.

**Comment by chembai** [comment]
ok thanks ,really appreciate the help

**Comment by chembai** [comment]
will installing kali linux v2.0 resolve all kind of issues?(wifi detection,low strength,connection,kernel modules)

**Comment by chembai** [comment]
it says unable to locate package kernel-package

**Comment by Bharat-Kalluri** [comment]
Hey @dean-brown , Can you please explain all the steps you have followed. I have had similar range and speed issues and have tried the master branch with no benefits. Please explain how to install the correct driver from the correct branch.

**Comment by lwfinger** [comment]
The master branch should now work. You need to do a 'git pull', "make", and 'sudo make install'. Once that is done, reboot. After the system is back up, you will need to

sudo modprobe -rv rtl8723be
sudo modprobe -v rtl8723be ant_sel=1

Then test. If that does not help, then use ant_sel=2 in that second command.

**Comment by lwfinger** [comment]
A completely new version of the drivers has just been pushed. It fixes a lot of things. As a result, I am closing all open issues. If you get a new one, please create a new issue.

Note: The new driver fixes the problems with power save. It is recommended that you clear all ips, swlps, and fwlps options.
